shared/dataflow/codeql/dataflow/internal/DataFlowImplCommon.qll

```diff
@@ -1642,19 +1643,15 @@ module MakeImplCommon<InputSig Lang> {
     }
   }
 
+  final private class NodeFinal = Node;
+
   /**
    * A `Node` at which a cast can occur such that the type should be checked.
    */
-  class CastingNode instanceof Node {
+  class CastingNode extends NodeFinal {
     CastingNode() { castingNode(this) }
 
     string toString() { result = super.toString() }
```

Comment:
When switching to `extends final` all of these `toString`s become superfluous, right?

Reply:
Right; I'll remove them.